### PR TITLE
Add custom Nunjucks environment for use in add-ons

### PR DESCRIPTION
**Pull request by allmarkedup**
This PR adds a new package to provide a customised, Fractal-specific Nunjucks environment for use in Fractal addons such as the inspector and static site builder.

### CODE REVIEW

File: packages/internals/funjucks/src/env.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    for (const filter of ['await', 'beautify', 'highlight', 'stringify']) {
+      it(`adds the ${filter} filter`, function () {
+        expect(() => env.getFilter(filter)).to.not.throw(Error);
+      });
+    }
+  });
+});
+
+function makeEnv() {
+  const fractal = {}; // TODO: Use proper Fractal instance
```
Comment (dkhuntrods) [inline packages/internals/funjucks/src/env.test.js]: I'd say a mock Fractal instance would be better here, can use a proper instance in the integration tests.

File: packages/internals/funjucks/src/env.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      const result = env.render('foo.html');
+      expect(result).to.be.instanceOf(Promise);
+      result.catch(err => {});
+    });
+
+    it('leaves other methods untouched', function () {
+      expect(env.addFilter('foo', () => {})).to.not.be.instanceOf(Promise);
+    });
+
+    it('adds the fractal instance as a property on the environment');
```
Comment (dkhuntrods) [inline packages/internals/funjucks/src/env.test.js]: Just checking this was intentionally left a stub.

File: packages/internals/funjucks/src/filters/await.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  it('exports a function', function () {
+    expect(factory).to.be.a('function');
+  });
+
+  it('returns an object with the expected properties', function () {
+    const filter = factory();
+    expect(filter.name).to.equal('await');
+    expect(filter.async).to.equal(true);
+    expect(typeof filter.filter).to.equal('function');
+  });
```
Comment (dkhuntrods) [inline packages/internals/funjucks/src/filters/await.test.js]: (Lines 5-14)
These look like they're pretty consistent across the filters, could be extracted to common filter tests?

File: packages/internals/funjucks/src/env.test.js (hunk 1 above)
Comment (allmarkedup) [inline packages/internals/funjucks/src/env.test.js]: Yeah i wasn't really sure of the best way to mock it (which is why the test above was also left as a stub). My intention was that when we have ported over the Fractal instance proper I can pull that in and do an `instanceof` check in the stubbed test. Is there a better way to handle this in the meantime?

File: packages/internals/funjucks/src/filters/await.test.js (hunk 3 above)
Comment (allmarkedup) [inline packages/internals/funjucks/src/filters/await.test.js]: Yes definitly, will be the same for the plugins, but I'd like to chat with you first to make sure we have a good re-usable pattern for it across the monorepo. Unless you have any objections I'll merge it in like this and we can come back to it once we have an good approach figured out.

File: packages/internals/funjucks/src/env.test.js (hunk 1 above)
Comment (dkhuntrods) [inline packages/internals/funjucks/src/env.test.js]: If there's direct functionality that the testee might require, a mock or stub with Sinon that would mock the functionality; if it's simply to check that that thing *is a* Fractal, and that's all, then what you've outlined is probably the simplest for now.
There are other options: https://github.com/sinonjs/sinon/issues/567#issuecomment-58648756

**Pull request merged**